vars/cloudFoundryDeploy.groovy

```diff
@@ -0,0 +1,148 @@
+import com.sap.piper.Utils
+import com.sap.piper.ConfigurationHelper
+
+import groovy.transform.Field
+
+@Field String STEP_NAME = 'cloudFoundryDeploy'
+@Field Set STEP_CONFIG_KEYS = ['cfApiEndpoint', 'cfAppName', 'cfCredentialsId', 'cfManifest', 'cfOrg', 'cfSpace', 'deployUser', 'deployTool', 'deployType', 'dockerImage', 'dockerWorkspace', 'mtaDeployParameters', 'mtaExtensionDescriptor', 'mtaPath', 'smokeTestScript', 'stashContent', 'useCAM']
+@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
+
+def call(Map parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        def utils = parameters.juStabUtils
+        if (utils == null) {
+            utils = new Utils()
+        }
+
+        def script = parameters.script
+        if (script == null)
+            script = [commonPipelineEnvironment: commonPipelineEnvironment]
+
+        Map config = ConfigurationHelper
+            .loadStepDefaults(this)
+            .mixinGeneralConfig(script.commonPipelineEnvironment, STEP_CONFIG_KEYS)
```

Comment:
What is the idea behind configuring step parameters in general

Reply:
as mentioned above, reasoning is that we share the same parameters across steps.
Examples are 
* cfSpace for testing
* cfApiEndpoint for testing
* credentialIds
* ...

Reply:
this is required for compatibility reasons